File: Source/DotNET/Model/Elements/Panels/DockPanel.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Scene.Model.Elements.Panels;

/// <summary>
/// Lays its children out against its edges, each child stating which edge through the <c>Dock</c> key
/// of its <see cref="SceneElement.Properties"/>.
/// </summary>
public record DockPanel : Panel
{
    /// <summary>
    /// Gets whether the last child spreads into whatever space the docked children left behind.
    /// </summary>
    public bool LastChildFill { get; init; } = true;
}

File: Source/JavaScript/model/elements/panels/DockPanel.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { Panel } from '../Panel';

/**
 * Lays its children out against its edges, each child stating which edge through the `Dock` key
 * of its `properties`.
 */
export interface DockPanel extends Panel {
    lastChildFill: boolean;
}

export const DockPanelPropertyNames: (keyof DockPanel)[] = ['lastChildFill'];
